Adding a code of conduct (#40)
2i2c-org/team-compass: culture/code-of-conduct.md

Threats
: threatening someone physically or verbally. For example, threatening to publicize sensitive information about someone’s personal life.

### Unwelcoming behavior

Blatant -isms
: saying things that are explicitly racist, sexist, homophobic, etc. For example, arguing that some people are less intelligent because of their gender, race or religion. {ref}`Subtle -isms <social:subtle-isms>` and small mistakes made in conversation are not code of conduct violations. However, repeating something after it has been pointed out to you that you broke a social rule, or antagonizing or arguing with someone who has pointed out your subtle -ism is considered unwelcoming behavior, and is not allowed at 2i2c.

Maliciousness towards other community members
: deliberately attempting to make others feel bad, name-calling, singling out others for derision or exclusion. For example, telling someone they’re not a real programmer or that they don’t belong at 2i2c.

Being especially unpleasant

> : for example, if we’ve received reports from multiple 2i2c users, team members, or collaborators of annoying, rude, or especially distracting behavior.

Reviewer: "annoying" raised a bit of a flag for me here. It places a lot of emphasis on the "receiver's" perception as opposed to the "actor's" intent. If someone ie eager and overly excited, that can be perceived as annoying by someone who is tired or over-worked. Though that is a situation to be addressed, I am not sure it should trigger a CoC violation. 

Does something like "agitating" capture the intent here?

Author: Hmmm I think `agitating` is a good word to use. I think that "annoying" may also be a bit of an over-loaded term with lots of connotations to particular personal characteristics.

Reviewer: I agree it is important to recognize that a CoC can be violated unintentionally, and that intent doesn't always align with the impacts of someone's actions. 

My concern here is more in-line your first point that "annoying" is a bit over-loaded and pretty easily attributed to a personal characteristics / differences in 2 personalities that might not align.

Author: For sure - I've added in `agitating` and removed `annoying` 👍

In addition, the 2i2c community and experience often extends outside those spaces—2i2c community members may go on walks together to get lunch, attend meetups or conferences as a group, communicate on social media, or interact with each other in other communities. Abusive or unwelcoming behavior between community members still has a profound impact on individuals and on the community when it happens beyond our walls. 2i2c Directors and the Steering Council will use our discretion when deciding whether to enforce this code of conduct and potentially remove someone from the 2i2c community after reports of such behavior happening outside of 2i2c, taking into account the impact on the individual community members involved as well as the impact on the community at large.

The 2i2c Code of Conduct does not apply to users of any hubs that we manage on behalf of other organizations, though we encourage leaders in those communities to adopt a Code of Conduct for their hub infrastructure. The Code of Conduct **does** apply to any 2i2c Team Member that is acting in a community's hub.

> When in doubt, please {ref}`report <coc:reporting>` unacceptable behavior to us. If someone’s behavior outside of a 2i2c space makes you feel unsafe at 2i2c, that is absolutely relevant and actionable for us.

Reviewer 2: nice. Would be nice to emphasize this a bit more.

## Enforcement

We’ve categorized unacceptable behavior into abuse and unwelcoming behavior in the section above.

> If we witness or receive a report about abusive behavior, we will contact the perpetrator to have a conversation with them and verify what has transpired, and they will be removed from the 2i2c community. Their Slack account will be deactivated, and permissions will be removed from any 2i2c-related repositories. They will not be welcome in any physical or digital spaces covered by the 2i2c Code of Conduct.

Reviewer: This makes it sound like removal of the perpetrator is the only possible outcome because it only discusses abusive behaviour and not unwelcoming behaviour as well. This is laid out nicely in the protocol doc, but if the CoC was all I read, it could easily be understood that the only possible outcome is removing someone (which might deter someone from submitting a CoC report on unwelcoming behaviour if they want to flag something, but not necessarily take drastic action). I think a short statement on unwelcoming behaviour here would solve that.

(coc:reporting)=
## Reporting

If you see a violation of our code of conduct, please [report it to the 2i2c Executive Director](https://docs.google.com/forms/d/e/1FAIpQLSeKFuoghGzftqqgaa0xllZyocdUJsWZmoSaFFlerimg4n_Y8A/viewform?usp=sf_link).

### Why should I report?

* **You are responsible for making 2i2c a safe and comfortable space for everyone.** Everyone in our community shares this responsibility. 2i2c Team and Steering Council members are not around the online spaces or at 2i2c events all the time, so we cannot enforce the code of conduct without your help.
* **The consequences for the 2i2c community of not reporting bad behavior outweigh the consequences for one person of reporting it.** We sometimes hear “I don’t want X person to meet consequences because I told someone about their bad behavior.” Consider the impact on everyone else at 2i2c of letting their behavior continue unchecked.
* 2i2c only works as an open, participatory, community-driven community because of shared trust between community members. **Reporting code of conduct violations helps us identify when this trust is broken, to prevent that from happening in the future.**

### Where and how to report

> Please report all code of conduct violations using [our reporting form](https://docs.google.com/forms/d/e/1FAIpQLSeKFuoghGzftqqgaa0xllZyocdUJsWZmoSaFFlerimg4n_Y8A/viewform?usp=sf_link). This is a short Google Form that will be sent to the Executive Director.

Reviewer: If you are comfortable with it, @choldgraf: I think it would be appropriate to include your email in this statement too. A form can feel a bit bureaucratic, but also having an email there signals that there is a human on the other end responsible for reading it.

Author: good point - perhaps we should make a generic address like `director@2i2c.org`?

If you would rather discuss the matter in person with a team member, send a direct message in the 2i2c Slack, or email a Steering Council member to schedule a time to talk.

### Confidentiality

We will keep all reports confidential, except if we've discussed with you and agreed otherwise. When we discuss incidents with people who are reported, we will anonymize details as much as we can to protect reporter privacy.

> However, some incidents happen in one-on-one interactions, and even if the details are anonymized, the reported person may be able to guess who made the report. If you have concerns about retaliation or your personal safety, and do not want us to share the details of your report with anyone (including the perpetrator) please let us know explicitly in your report. Unfortunately, in that situation we will not be able to take any action.

Reviewer 2: Hmm, does the recurse center coc say that they won't be able to take any action? how do other communities deal with this? This seems difficult but very important.

Author: Yeah this one is from their document. The one place where I have seen this not be true is with Title IX stuff in universities, but not sure if that applies to us. I think the question is whether there are some cases where we would take action even if the person explicitly asked us not to.

In some cases we may decide to share an update about a major incident with 2i2c team members, or with the entire 2i2c community. If that's the case, the identities of all victims and reporters will remain confidential unless those individuals instruct us otherwise.

## Social rules

> In addition to having a code of conduct, we have four lightweight [social rules](social-rules.md). The social rules are different and separate from the code of conduct. They help us create a better learning environment by giving names to counterproductive behavior and acting as a release valve so that frustration doesn't build up over time. We expect people to unintentionally break the social rules from time to time. Doing this doesn't make you a bad person or a bad community member. When this happens, it's not a big deal. Just apologize and move on.

Reviewer: ```suggestion
In addition to having a code of conduct, we have four lightweight [social rules](social-rules.md). The social rules are different and separate from the code of conduct. They help us create a better learning environment by giving names to counterproductive behavior and acting as a release valve so that frustration doesn't build up over time. We understand and anticipate people to unintentionally break the social rules from time to time. Doing this doesn't make you a bad person or a bad community member. When this happens, it's not a big deal. Just apologize and move on.
```

Reviewer: (don't want "expect" to be confused with "expectation" -- we hope people don't break these rules too often :-)

> The enforcement provisions in this code of conduct do not apply to the social rules. We definitely won't give you a strong warning or expel you from the 2i2c community just for breaking a social rule.

Reviewer: ```suggestion
The enforcement provisions in this code of conduct do not apply to the social rules. We won't give you a strong warning or expel you from the 2i2c community just for breaking a social rule.
```

> If you have any questions about any part of the code of conduct or social rules, please reach out to any 2i2c team member.

Reviewer: is there a url here we could add for 2i2c/people ? showing who those people are help make it a bit more human

Author: good point, I'll link to the `About Us` page for now

## How we developed the code of conduct

We arrived at these policies by a combination of:

* Listening to feedback and suggestions we've heard from open source communities over many years
* Reading the codes of conduct of other organizations we find to be thoughtful (see some examples below)
* Considering our experiences in other communities and projects in the past

## Other things that don’t fit in to the code of conduct

### When to seek help immediately

Instead of filling out a code of conduct violation report, please contact law enforcement directly to report criminal activity (e.g. physical assault, sexual assault, theft), or to report a dangerous physical situation (e.g. fire, serious injury, fear that someone will hurt themselves or someone else).

### Getting help

> If you or someone else at 2i2c is struggling and needs help, don’t hesitate to reach out to the 2i2c Directors and Steering Council members in person or over email.

Reviewer: contact info here too?

:hidden:
response-protocol
```

## Why have a code of conduct?

Our goal is to create the best community in the world for learning, using, and creating open infrastructure for interactive computing.
We want every member of the 2i2c community to be able to focus their full attention on building amazing tools and developing a thriving culture of collaboration and inclusion around them.
This is impossible to do if you are being harassed, stalked, or discriminated against.

Accordingly, anyone that participates in a 2i2c space is expected to show respect and courtesy to each other in all interactions, whether in GitHub repositories, our Slack channel, in in-person events, when representing 2i2c in public, or in events and spaces associated with {term}`ICSI`.

To make sure that everyone has a common understanding of “show respect and courtesy to each other,” we have adopted the following code of conduct. The code of conduct is enforced by the 2i2c Executive Director and the Steering Council.

## Unacceptable behavior

The following types of behavior are unacceptable in 2i2c spaces, both online and in-person, and constitute code of conduct violations.

### Abusive behavior

Harassment

> : including offensive verbal comments related to gender, sexual orientation, disability, physical appearance, body size, race, or religion, as well as sexual images in public spaces, deliberate intimidation, stalking, following, harassing photography or recording, inappropriate physical contact, and unwelcome sexual or romantic attention.

Reviewer 3: ```suggestion
: including offensive verbal comments related to gender, sexual orientation, disability, physical appearance, body size, race, religion, or political views; as well as sexual images in public spaces, deliberate intimidation, stalking, following, harassing photography or recording, inappropriate physical contact, and unwelcome sexual or romantic attention.
```

I suggest adding "political views" to this list. I did this in our research group's [code of conduct](https://ocean-transport.github.io/guide.html). The logic here is to try to be welcoming to folks who don't necessary conform with the political orthodoxy that has become ubiquitous within academic spaces. Perhaps this is controversial, but I like to think that a person with conservative political views would be welcome at any organization I'm part of, even if I disagree with them.

Threats
: threatening someone physically or verbally. For example, threatening to publicize sensitive information about someone’s personal life.

### Unwelcoming behavior

Blatant -isms

> : saying things that are explicitly racist, sexist, homophobic, etc. For example, arguing that some people are less intelligent because of their gender, race or religion. {ref}`Subtle -isms <social:subtle-isms>` and small mistakes made in conversation are not code of conduct violations. However, repeating something after it has been pointed out to you that you broke a social rule, or antagonizing or arguing with someone who has pointed out your subtle -ism is considered unwelcoming behavior, and is not allowed at 2i2c.

Reviewer 3: I would prefer not to use "etc." here. Can we explicitly enumerate the "isms" that constitute unwelcoming behavior? Ambiguity opens the door to this being applied inconsistently and unfairly.

Maliciousness towards other community members

> : deliberately attempting to make others feel bad, name-calling, singling out others for derision or exclusion. For example, telling someone they’re not a real programmer or that they don’t belong at 2i2c.

Reviewer 3: Who determines if comments are deliberately malicious vs. unintentionally. The use of "deliberate" means that we care about the offender's intentions? Do we? Or just the impact. All of the harassment training I have received emphasizes impact over intention.  Based on that, I question whether this list item is necessary.

Being especially unpleasant

> : for example, if we’ve received reports from multiple 2i2c users, team members, or collaborators of agitating, rude, or especially distracting behavior over an extended period of time.

Reviewer 3: Again, this feels vague. "Especially" implies a comparison to some baseline. What is the baseline?

The 2i2c Code of Conduct does not apply to users of any hubs that we manage on behalf of other organizations, though we encourage leaders in those communities to adopt a Code of Conduct for their hub infrastructure. The Code of Conduct **does** apply to any 2i2c Team Member that is acting in a community's hub.

:::{important}
When in doubt, please {ref}`report <coc:reporting>` unacceptable behavior to us. If someone’s behavior outside of a 2i2c space makes you feel unsafe at 2i2c, that is absolutely relevant and actionable for us.
:::

## Enforcement

We’ve categorized unacceptable behavior into abuse and unwelcoming behavior in the section above.

If we witness or receive a report about abusive behavior, we will contact the perpetrator to have a conversation with them and verify what has transpired. We will [follow this response protocol](response-protocol.md).

> If we verify abusive behavior, they will be removed from the 2i2c community. Their Slack account will be deactivated, and permissions will be removed from any 2i2c-related repositories. They will not be welcome in any physical or digital spaces covered by the 2i2c Code of Conduct.

Reviewer 3: What if the person works for ICSI? Will they be fired?

(coc:reporting)=
## Reporting

If you see a violation of our code of conduct, please [report it to the 2i2c Executive Director](https://docs.google.com/forms/d/e/1FAIpQLSeKFuoghGzftqqgaa0xllZyocdUJsWZmoSaFFlerimg4n_Y8A/viewform?usp=sf_link).

### Why should I report?

* **You are responsible for making 2i2c a safe and comfortable space for everyone.** Everyone in our community shares this responsibility. 2i2c Team and Steering Council members are not around the online spaces or at 2i2c events all the time, so we cannot enforce the code of conduct without your help.
* **The consequences for the 2i2c community of not reporting bad behavior outweigh the consequences for one person of reporting it.** We sometimes hear “I don’t want X person to meet consequences because I told someone about their bad behavior.” Consider the impact on everyone else at 2i2c of letting their behavior continue unchecked.
* 2i2c only works as an open, participatory, community-driven community because of shared trust between community members. **Reporting code of conduct violations helps us identify when this trust is broken, to prevent that from happening in the future.**

### Where and how to report

> Please report all code of conduct violations using [our reporting form](https://docs.google.com/forms/d/e/1FAIpQLSeKFuoghGzftqqgaa0xllZyocdUJsWZmoSaFFlerimg4n_Y8A/viewform?usp=sf_link). This is a short Google Form that will be sent to the Executive Director. If you prefer, you may also [email the Executive Director directly](mailto:director@2i2c.org).

Reviewer 3: Not that I expect this to happen, but we should probably have an alternative channel in case the incident itself involves the ED.

Reviewer 4: I don't have a good solution here but I agree with Ryan. An incident involving the ED needs a reporting path. I have concerns about a different aspect though. The ED is naturally incentivized to protect and sustain 2i2c and there may emerge situations where that goal is in tension with proper resolution on an incident. Imagine some situations involving a talented 2i2c developer and a community member.... I can imagine an abstract ED (not Chris mind you...) choosing, perhaps wishing is a better word here, to minimize the incident to retain the developer leaving the community member harmed and with justice not being served. At a minimum, two people should receive the reports.

In situations where 2i2c may be serving another community, (e.g. OpenScapes using 2i2c infrastructure), we should find a way to integrate the organizers/leaders of the other community in the CoC management. Ideally, this integration should also build in the expectation that participants actually read the CoC document and behave accordingly. 

I was impressed by the way ADSA handled CoC.
